File: buildSrc/src/main/java/fr/pgreze/context/ContextPlugin.java
import org.gradle.api.Plugin;
import org.gradle.api.Project;

import java.net.InetAddress;
import java.net.UnknownHostException;
import java.util.Date;

public class ContextPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
        project.getTasks().create("context", ContextPrinter.class, task -> {
            task.setNow(new Date());
            task.setHostname(getHostname());
        });
    }

    private String getHostname() {
        try {
            return InetAddress.getLocalHost().getHostName();
        } catch (UnknownHostException e) {
            return null;
        }
    }
}

File: buildSrc/src/main/java/fr/pgreze/context/ContextPlugin.kt
package fr.pgreze.context

import org.gradle.api.Plugin
import org.gradle.api.Project
import java.net.InetAddress
import java.net.UnknownHostException
import java.util.Date

open class ContextPlugin : Plugin<Project> {
    override fun apply(project: Project) {
        project.tasks.register("context", ContextPrinter::class.java) {
            now = Date()
            hostname = try {
                InetAddress.getLocalHost().hostName
            } catch (e: UnknownHostException) {
                null
            }
        }
    }
}
